src/groovy/com/recomdata/security/LdapAuthUserDetailsMapper.groovy

```diff
@@ -134,11 +157,19 @@ public class LdapAuthUserDetailsMapper implements UserDetailsContextMapper {
                 "use a subclass if mapUserToContext() is required.");
     }
 
-    protected String mapPassword(Object passwordValue) {
-        if (!(passwordValue instanceof String)) {
-            passwordValue = new String((byte[]) passwordValue);
+    protected String mapPassword(DirContextOperations ctx) {
+        if (noPassword) {
+            return 'NO_PASSWORD'
+        }
+        String password = ''
+        Object passwordValue = ctx.getObjectAttribute("userPasswordRaw");
+        if (passwordValue != null) {
+            if (!(passwordValue instanceof String)) {
+                passwordValue = new String((byte[]) passwordValue);
+            }
+            password = (String) passwordValue
         }
-        return (String) passwordValue;
+        return springSecurityService.encodePassword(password)
```

Comment:
I'm not sure how it actually work and what is "userPasswordRaw" mean. I just extracted existing code to separate method.

Old code:
```groovy
protected String mapPassword(Object passwordValue) {
        if (!(passwordValue instanceof String)) {
            passwordValue = new String((byte[]) passwordValue);
        }
        return (String) passwordValue;
    }

Object passwordValue = ctx.getObjectAttribute("userPasswordRaw");

        if (passwordValue != null) {
            password = mapPassword(passwordValue);
        }

if (!user) {
                create = true
                def sql = new Sql(dataSource)
                def seqSQL = databasePortabilityService.getNextSequenceValueSql('searchapp', 'hibernate_sequence')
                def result = sql.firstRow(seqSQL)
                user = new AuthUser(id: result.nextval, username: username, passwd: springSecurityService.encodePassword(password), name: fullName, userRealName: fullName, email: email, emailShow: true, enabled: true)
            } else {
                user.passwd = springSecurityService.encodePassword(password)
                user.userRealName = fullName
                user.email = email
            }
```

`encodePassword` will not ever called with `null` value, because it initializes `password` with empty string and reassigns it only if `passwordValue` is not null.

I afraid touch existing logic, because it may break already working (?) functionality.

I renamed `noPassword` to `inheritPassword` and added default `true` to Config.groovy.